plugin-lifecycle/src/main/java/com/mapbox/maps/plugin/lifecycle/ViewLifecycleOwner.kt

```diff
@@ -0,0 +1,95 @@
+package com.mapbox.maps.plugin.lifecycle
+
+import android.view.View
+import androidx.annotation.UiThread
+import androidx.annotation.VisibleForTesting
+import androidx.annotation.VisibleForTesting.PRIVATE
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleEventObserver
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.LifecycleRegistry
+import androidx.lifecycle.ViewTreeLifecycleOwner
+import java.lang.ref.WeakReference
+import java.util.concurrent.Flow
+
+/**
+ * A [LifecycleOwner] that merges the hosting lifecycle owner's [Lifecycle] (of an Activity or Fragment)
+ * with the [View]'s drawing surface lifecycle.
+ *
+ * This lifecycle is useful for continuously running tasks that obtain the data and draw it in the [View],
+ * for example by collecting [Flow]s and drawing results in the nested views or on the map.
+ *
+ * In general, this lifecycle reflects the hosting lifecycle owner's [Lifecycle] whenever the `View` is attached
+ * but whenever the `View` is detached and we want to stop the coroutines tied to the lifecycle's scope,
+ * this lifecycle will transition to [Lifecycle.State.CREATED] (and invoke [Lifecycle.Event.ON_STOP])
```

Comment:
Looked in the [onStop](https://github.com/mapbox/mapbox-maps-android/blob/main/sdk/src/main/java/com/mapbox/maps/MapController.kt#L144) method, couldn't we have an issue if view was detached, then onStyleLoaded event arrives and plugin dont receive the updated one since they've unsubscribed? 


```
    this.onStyleDataLoadedListener = OnStyleDataLoadedListener { eventData ->
      if (eventData.type == StyleDataType.STYLE) {
        mapboxMap.getStyle { style ->
          pluginRegistry.onStyleChanged(style)
        }
      }
    }
```

Reply:
🤔 in this case, if the mapview is detached, do we still want to get the callbacks? I think all the plugin is aware of the onStart/onStop lifecycle events and should be prepared for re-initialisation during `onStart`?

Reply:
@yunikkk we don't know if the style loaded event is received before the MapView's onStop, so I'd rather prefer to initialise the plugin in onStart and do proper clean up in onStop. And whenever the style is needed to initialise the plugin, we should get the style using `MapboxMap.getStyle { style -> }`, wdyt?

Reply:
I was mainly concerned that triggering `onStyleChanged` manually on every `onStart` might result in adding the same layer/source to the same style repeatedly, and result in crash. As the documentation of the `onStyleChanged` suggest that it will only be called when a new Style is loaded, users may under the impression they need to redo all the style related changes.

Just checked code base and the plugins that extends `MapStyleObserverPlugin`, indeed it seems we are only updating the style reference but not triggering any other operations, so I guess it's safe to trigger `onStyleChanged` with our current code base, although it might still a bit confusing for the future plugin developers.

Reply:
Having said that, I think it's still a valid issue that we need to think more about, @yunikkk could you cut a new ticket for it and we can address it in a separate PR?

Reply:
Added the ticket